src/Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native/SoftwareEntitlementClient.cpp

```diff
@@ -389,6 +390,82 @@ class Curl
 
         return CURLE_OK;
     }
+
+    struct WinHttpDeleter
+    {
+        void operator()(HINTERNET h)
+        {
+            WinHttpCloseHandle(h);
+        }
+    };
+
+    typedef std::unique_ptr<void, WinHttpDeleter> WinHttpHandle;
+
+    //
+    // OpenSSL does not hook into the Windows Automatic Root Certificates Update process.
+    // This results in certificate validation failures, so we perform a dummy connection
+    // using Windows SChannel which will setup the root cert store correctly.
+    //
+    static void EnsureRootCertsArePopulated(const std::string& url)
+    {
+        std::string prefix("https://");
+        size_t start = url.rfind(prefix, 0);
+        if (start == std::string::npos)
```

Comment:
Could this be spoofed by including a parameter at the end of the URL like `....?querytext="https://"` ?

I'm wondering whether it would be better to check that it's *only* found at the start of the string.

Reply:
that's what `rfind` does - searches backwards, and I tell it to start at the beginning.  So if it's not found starting at 0, it won't be found at all.